syntax = "proto3";

package task;

service TaskService {
    rpc AddTask (TaskRequest) returns (TaskResponse);
    rpc GetTask (TaskId) returns (Task);
    rpc GetAllTasks (Empty) returns (TaskList);
}

message TaskRequest {
    string title = 1;
    string description = 2;
}

message TaskResponse {
    string message = 1;
}

message TaskId {
    string id = 1;
}

message Task {
    string id = 1;
    string title = 2;
    string description = 3;
}

message TaskList {
    repeated Task tasks = 1;
}

message Empty {}
